File: src/com/epi/problems/linkedlists/Runner.java
package com.epi.problems.linkedlists;

public class Runner {

    public static void main(String[] args){
        LinkedList<Integer> first = LinkedList.getList(new Integer[]{1,2,3,4,5,6,78,80});
//        LinkedList.addCycle(first, 7);
        LinkedList.ListNode<Integer> cycleStart = ProblemSolver.findCycle(first);
        System.out.print(cycleStart == null ? "No cycle" : cycleStart.data);
    }
}

File: src/com/epi/problems/search/Runner.kt
package com.epi.problems.search

fun main(){

    print(ProblemSolver.findRotationPoint(arrayOf(1,2,3,4,5,6)))
    print(ProblemSolver.findRotationPoint(arrayOf(6,1,2,3,4,5)))
    print(ProblemSolver.findRotationPoint(arrayOf(5,6,1,2,3,4)))
    print(ProblemSolver.findRotationPoint(arrayOf(4,5,6,1,2,3)))
    print(ProblemSolver.findRotationPoint(arrayOf(3,4,5,6,1,2)))
    print(ProblemSolver.findRotationPoint(arrayOf(2,3,4,5,6,1)))
}